plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -339,15 +340,36 @@ public Table loadTable(ConnectorSession session, SchemaTableName table)
     @Override
     public void dropTable(ConnectorSession session, SchemaTableName schemaTableName)
     {
-        BaseTable table = (BaseTable) loadTable(session, schemaTableName);
-        validateTableCanBeDropped(table);
+        BaseTable table = null;
+        try {
+            table = (BaseTable) loadTable(session, schemaTableName);
+            validateTableCanBeDropped(table);
+        }
+        catch (Exception e) {
+            if (!(e.getCause() instanceof FileNotFoundException)) {
+                throw e;
+            }
+            LOG.warn("Failed to load table " + schemaTableName, e);
+        }
+
         try {
             deleteTable(schemaTableName.getSchemaName(), schemaTableName.getTableName());
         }
         catch (AmazonServiceException e) {
             throw new TrinoException(HIVE_METASTORE_ERROR, e);
         }
-        dropTableData(table.io(), table.operations().current());
+
+        if (table != null) {
+            try {
+                dropTableData(table.io(), table.operations().current());
+            }
+            catch (Exception e) {
+                if (!(e.getCause() instanceof FileNotFoundException)) {
+                    throw e;
+                }
+                LOG.warn("Failed to delete data files referenced by TableMetadata for table " + schemaTableName, e);
```

Comment:
Same situation here, why is the exception being swallowed?

Reply:
Because we do not want to fail when FileNotFoundException exception occurs instead we want to drop the table from metastore and delete table location directory.